src/agent/agent.js

```diff
@@ -576,47 +577,70 @@ class Agent {
             }
             this.server = https_server;
         } else if (addr_url.protocol === 'tcp:') {
-            const tcp_server = this.rpc.register_tcp_transport(addr_url.port);
+            const tcp_server = this.rpc.create_tcp_server(addr_url.port);
             tcp_server.on('close', () => {
                 dbg.warn('AGENT TCP SERVER CLOSED');
                 retry();
             });
+            tcp_server.listen(addr_url.port).catch(err => {
+                dbg.warn('AGENT TCP SERVER ERROR', err.stack || err);
+                retry();
+            });
             this.server = tcp_server;
         } else if (addr_url.protocol === 'tls:') {
-            const tls_server = this.rpc.register_tcp_transport(addr_url.port, this.ssl_context);
+            const tls_server = this.rpc.create_tcp_server(addr_url.port, this.ssl_context);
             tls_server.on('close', () => {
                 dbg.warn('AGENT TLS SERVER CLOSED');
                 retry();
             });
+            tls_server.listen(addr_url.port).catch(err => {
+                dbg.warn('AGENT TLS SERVER ERROR', err.stack || err);
```

Comment:
hmm I guess because it's not a fatal case, and it keeps retrying. but it could be an error too. I don't think there is a clear distinction as to what should be warn vs error, both are visible enough.